cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/TemplateSetConfiguration.java

```diff
@@ -0,0 +1,284 @@
+
+    this.triggers = Maps.newHashMap();
+    this.templates = Maps.newHashMap();
+    this.rootTemplateFolders = Maps.newHashMap();
+    this.utilFolders = Maps.newHashMap();
+    this.configRoot = configurationPath;
+    readConfiguration(configurationPath);
+  }
+
+  /**
+   * Reads the configuration from the given path
+   *
+   * @param configurationPath CobiGen configuration root path
+   * @throws InvalidConfigurationException thrown if the {@link File} is not valid with respect to the context.xsd
+   */
+  public void readConfiguration(Path configurationPath) throws InvalidConfigurationException {
+
+    if (this.templateSetConfigurationReader == null) {
+      this.templateSetConfigurationReader = new TemplateSetConfigurationReader(configurationPath);
+    }
+
+    List<Path> templateSetsAdapted = this.templateSetConfigurationReader.getTemplateSetConfigurationPathsAdapted();
+    List<Path> templateSetsDownloaded = this.templateSetConfigurationReader
+        .getTemplateSetConfigurationPathsDownloaded();
+
+    this.increments = new HashMap<>();
+    this.templatesConfigurations = Lists.newLinkedList();
+    if (!templateSetsAdapted.isEmpty()) {
+      for (Path templateSetFile : templateSetsAdapted) {
+        initializeTemplateSets(false, configurationPath, templateSetFile);
+      }
+    }
+
+    if (!templateSetsDownloaded.isEmpty()) {
+      for (Path templateSetFile : templateSetsDownloaded) {
+        initializeTemplateSets(true, configurationPath, templateSetFile);
+      }
+    }
+
+  }
+
+  /**
+   * Initializes template sets
+   *
+   * @param isZipFile boolean true if downloaded template-sets need to be processed
+   * @param configurationPath CobiGen configuration root path
+   * @param templateSetFile Path to template-set xml to be processed
+   */
+  private void initializeTemplateSets(boolean isZipFile, Path configurationPath, Path templateSetFile) {
+
+    this.templateSetConfigurationReader.readConfiguration(templateSetFile);
+
+    com.devonfw.cobigen.impl.config.entity.io.TemplatesConfiguration templatesConfigurationStatic = this.templateSetConfigurationReader
+        .getTemplatesConfiguration();
+
+    com.devonfw.cobigen.impl.config.entity.io.ContextConfiguration contextConfigurationStatic = this.templateSetConfigurationReader
+        .getContextConfiguration();
+
+    TemplateFolder templateFolder = this.templateSetConfigurationReader.getRootTemplateFolder();
+
+    TemplatesConfigurationReader templatesConfigurationReader = new TemplatesConfigurationReader(
+        templatesConfigurationStatic, templateFolder, this.configurationHolder, templateSetFile);
+
+    ContextConfigurationReader contextConfigurationReader = new ContextConfigurationReader(contextConfigurationStatic,
+        templateSetFile);
+
+    Map<String, Trigger> trigger = contextConfigurationReader.loadTriggers();
+    Trigger activeTrigger = trigger.get(trigger.keySet().toArray()[0]);
```

Comment:
Add a comment why we take only the 1st trigger (because a template set has only one trigger)

Reply:
added.